README.md

```diff
@@ -49,51 +49,59 @@ pkg:docker/philips-labs/fatt@sha256:6cc65b2c82c2baa3391890abb8ab741efbcbc87baff3
-  The following checks were performed on each of these signatures:
-    - The cosign claims were validated
-    - The signatures were verified against the specified public key
-
-  $ while read -r a ; do sget "$a" ; done <<< "$attestations"
-  {
-    "SPDXID": "SPDXRef-DOCUMENT",
-    "name": "ghcr.io/philips-labs/slsa-provenance-v0.7.2",
-    "spdxVersion": "SPDX-2.2",
-    "creationInfo": {
-      "created": "2022-02-25T13:01:35.3837117Z",
-      "creators": [
-        "Organization: Anchore, Inc",
-        "Tool: syft-0.38.0"
-      ],
-      "licenseListVersion": "3.16"
-    },
-    …
-  ```
-
-</details>
+## OCI as blob storage
+
+Using OCI as a blob storage we can leverage this to distribute and discover attestations for software assets that are not OCI/Docker images. For this ecosystem we already have nice integration using [cosign](https://github.com/sigstore/cosign). With `fatt` we expand these capabilities to also support other ecosystems like NPM, Nuget, Gradle. Based on a publishing convention we can make it clear for our package consumers where they can retrieve our `build provenance` and `SBOM`.
+
+Any attestations published to an OCI registry should be captured in a `attestations.txt`. Fatt publish command automates the manual steps to upload the attestations using cosign and then generates an `attestations.txt` using PURL references of the published attestations. Then it also automatically publishes these `attestations.txt`. Manually you would perform following steps:
+
+```shell
+$ tree .
+.
+├── provenance.att
+└── sbom-spdx.json
+$ cosign upload blob -f provenance.att ghcr.io/philips-labs/fatt-attestations-example:v0.1.0.provenance
+$ cosign sign --key cosign.key ghcr.io/philips-labs/fatt-attestations-example:v0.1.0.provenance
+$ cosign upload blob -f sbom-spdx.json ghcr.io/philips-labs/fatt-attestations-example:v0.1.0.sbom
+$ cosign sign --key cosign.key ghcr.io/philips-labs/fatt-attestations-example:v0.1.0.sbom
+# Now you would manually fill out an attestations.txt using these image digests following this spec
+# https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#oci
+$ tree
+.
+├── attestations.txt
+├── provenance.att
+└── sbom-spdx.json
+$ cosign upload blob -f attestations.txt ghcr.io/philips-labs/fatt-attestations-example:v0.1.0.discover
+$ cosign sign --key cosign.key ghcr.io/philips-labs/fatt-attestations-example:v0.1.0.discover
+# Now we have the discovery file stored in an OCI registry we can look it up by digest
+$ fatt list --key cosign.pub ghcr.io/philips-labs/fatt-attestations-example:v0.1.0
+```
+
+See below how `fatt publish` automates this whole flow for you. (:warning: NOTE that signing currently is not yet build in. #20)
+
+```shell
+$ tree .
+.
+├── provenance.att
+└── sbom-spdx.json
+$ fatt publish --repository ghcr.io/philips-labs/fatt-attestations-example --version v0.1.0 sbom://sbom-spdx.json provenance://provenance.att
```

Comment:
I don't think it's clear to the reader on where the ``sbom://`` and ``provenance://`` comes from. Can you clarify this part please?
Could be as simple as: "Attestations are prefixed with the attestation type to make it possible for Fatt to build the correct PURL. Examples are ``sbom://`` or ``provenance://``"

Reply:
will update to introduce this in the docs here, FYI the cli gives an nice error